orchagent/portsorch.cpp

```diff
@@ -2351,6 +2351,22 @@ void PortsOrch::doLagMemberTask(Consumer &consumer)
     }
 }
 
+void PortsOrch::doTask()
```

Comment:
>doTask [](start = 16, length = 6)

Why you override Orch::doTask()? It will leave tasks in other tables undone.

Reply:
Which tables? everything that m_consumerMap has will be processed

Reply:
I see you are trying to specify drain order. How about reuse `Orch::doTask` for your second half?

---
In reply to: [339998153](https://github.com/Azure/sonic-swss/pull/1108#discussion_r339998153) [](ancestors = 339998153)

Reply:
If I reuse Orch::doTask failed items from APP_PORT_TABLE will be processed again, thats why a condition ```if (consumer == portConsumer)``` to skip port table was added in second half.

Another scenario - when  portsorch processes "PortInitDone" it explicitly returns because other orchs can start their operations. Orch::doTask reuse will make PortsOrch proceed after PortinitDone and configure port attributes like admin status, mtu, speed and so on, before BufferOrch  apply buffers so there will be a chance that ports will be operationaly up when buffers are applied.